src/test/ui/nll/issue-51191.rs

```diff
@@ -1,13 +1,15 @@
+#![allow(unconditional_recursion)]
+
```

Comment:
Any reason we're allowing this lint? Seems like it's not really related to the rest of the changes.

Reply:
An error about an unexcepted help message showed up related to the unconditional recursion after I added the help message changes, I can revert it and add the error annotation if you want.

Reply:
That would be great, thanks!

Reply:
Fixed.